libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -109,10 +110,7 @@ bool ByteBuffer::advance(uint32_t n)
     return true;
 }
 
-/*
-  read len bytes without advancing the read pointer
- */
-uint32_t ByteBuffer::peekbytes(uint8_t *data, uint32_t len)
+int ByteBuffer::peekiovec(ByteBuffer::IoVec iovec[2], uint32_t len)
```

Comment:
That was also true in the old code, but since it has less calls, I think it was being less affected, but still vulnerable.